core/src/main/java/io/questdb/griffin/engine/functions/eq/EqVarcharFunctionFactory.java

```diff
@@ -65,42 +67,68 @@ public Function newInstance(
         final Function a = args.getQuick(0);
         final Function b = args.getQuick(1);
 
-        if (a.isConstant() && !b.isConstant()) {
+        if (isAnytimeConstant(a) && !isAnytimeConstant(b)) {
             return createHalfConstantFunc(a, b);
         }
 
-        if (!a.isConstant() && b.isConstant()) {
+        if (!isAnytimeConstant(a) && isAnytimeConstant(b)) {
             return createHalfConstantFunc(b, a);
         }
 
         return new Func(a, b);
     }
 
+    private static boolean isAnytimeConstant(Function f) {
+        return f.isConstant() || f.isRuntimeConstant();
+    }
+
     private Function createHalfConstantFunc(Function constFunc, Function varFunc) {
-        Utf8Sequence constValue = constFunc.getVarcharA(null);
-        if (constValue == null) {
-            return new NullCheckFunc(varFunc);
+        if (constFunc.isConstant()) {
+            Utf8Sequence constValue = constFunc.getVarcharA(null);
+            if (constValue == null) {
+                return new NullCheckFunc(varFunc);
+            }
+            return new ConstCheckFunc(varFunc, constValue);
         }
-        return new ConstCheckFunc(varFunc, constValue);
+        return new ConstCheckFunc(varFunc, constFunc);
     }
 
     public static class ConstCheckFunc extends NegatableBooleanFunction implements UnaryFunction {
         private final Function arg;
-        private final Utf8Sequence constant;
+        private Function constFunc;
+        private Utf8Sequence constant;
 
         public ConstCheckFunc(Function arg, Utf8Sequence constant) {
             this.arg = arg;
+            this.constFunc = null;
             this.constant = constant;
         }
 
+        public ConstCheckFunc(Function varFunc, Function constFunc) {
+            this.arg = varFunc;
+            this.constFunc = constFunc;
+        }
+
+        @Override
+        public void init(SymbolTableSource symbolTableSource, SqlExecutionContext executionContext) throws SqlException {
+            UnaryFunction.super.init(symbolTableSource, executionContext);
+            if (constFunc != null) {
+                constant = constFunc.getVarcharA(null);
+                constFunc = null;
```

Comment:
`init()` is called before each query execution, so we shouldn't erase `constFunc`. If the bind variable value changes in subsequent prepared statement executions, the change won't be applied.

nit: the code may be cleaner with two separate function classes: one for the "normal" constant case and another one for the runtime constant case.

Reply:
The reason I used a single class is this:

```java
        if (isAnytimeConstant(a) && !isAnytimeConstant(b)) {
            return createHalfConstantFunc(a, b);
        }

        if (!isAnytimeConstant(a) && isAnytimeConstant(b)) {
            return createHalfConstantFunc(b, a);
        }
```

There would be an explosion of combinations with two classes.